### DISCUSSION TITLE
Issue with Stripe API key

### DISCUSSION THREAD

**Discussion by virajsoni06** [swellstores/community#478] [Help] [answered]
Hi, we have integrated Stripe for payments. It worked fine in test mode but in live mode when we try to make a payment we get the error that the platform access is revoked. We even tried to reconnect Stripe and still the same results.
Another strange thing is the `pk_live_` key in the screenshot doesn't match the key in Stripe

![image](https://github.com/swellstores/community/assets/13236286/9436cca8-99ad-4006-a164-dd15812dc978)

**Comment by virajsoni06** [comment]
@logeshswell bumping this up

**Comment by virajsoni06** [comment]
Hi @steven-k1 @ericingram, can someone get back on this with urgency. Our entire deployment is on hold because of this payment issue.

**Reply by logeshswell** [reply]
Let me have a look at it now.

**Reply by logeshswell** [reply]
Can you share your store URL?

**Reply by virajsoni06** [reply]
Store is setup here -> https://exambeacon.co.uk/
Try to add a product and go through the checkout, it will fail when using Stripe payments

**Reply by virajsoni06** [reply]
Our Swell store id is topmarks-test-store

**Reply by logeshswell** [reply]
Thanks for the information. I will check and get back to you very shortly.

**Reply by virajsoni06** [reply]
Thanks, will wait for your response

**Reply by logeshswell** [reply]
Do you see this public key: pk_live_51Ng****Q4QG in the error message?
Because Swell system contains the above live public key under Stripe payment settings.

**Reply by virajsoni06** [reply]
It contains `pk_live_***m1nOPo` but that's not the public key we have on Stripe

**Reply by virajsoni06** [reply]
Actually, right now I do see the `pk_live_51Ng****Q4QG` key, but previously I was seeing the above key. Regardless even `pk_live_51Ng****Q4QG` is not the key in our Stripe account

**Reply by logeshswell** [reply]
Understood. So, it is confirmed that the Swell dashboard contains/passes the right public key to the Stripe system.

**Reply by logeshswell** [reply]
How do you say that it's not the right key in your Stripe account?

Can you cross-check the response from the below URL to verify that you connected the right Stripe account in Swell dashboard?
https://topmarks-test-store.swell.store/admin/console?get/settings/-payments~Hgateways~Hstripe

**Reply by logeshswell** [reply]
Does `live_account_id` and `live_secret_key` matches with your Stripe account?

**Reply by virajsoni06** [reply]
@logeshswell No it doesn't match the ones we have in our Stripe account

**Reply by virajsoni06** [reply]
@logeshswell what's the update on this?

**Comment by steven-k1** [comment]
@virajsoni06 If the `live_account_id` and `live_secret_key` in Swell do not match up, it sounds like at some point, a different Stripe account was connected, so the system is attempting to connect to account B when you need to connect to account A.

The `live_account_id` and `live_secret_key` in Swell need to match up with the ones in the account you want to connect with in Stripe.

**Reply by virajsoni06** [reply]
@steven-k1 even if account B is connected Swell should be updating the keys on their system. There's no input for the store owners to change any Stripe integration stuff. This is clearly a bug in Swell's Stripe integration. Can you get this fixed ASAP! Or atleast provide a reliable workaround to fix it on our store

**Reply by ericingram** [reply]
Looking into it

**Reply by logeshswell** [reply]
I disconnected the Stripe integration from Swell admin dashboard. Can you try connecting Swell with the correct Stripe account?
Before that, please try logout and then log in using the correct Stripe account in the Stripe dashboard.

**Reply by virajsoni06** [reply]
@logeshswell @ericingram still doesn't work. They keys shown in console doesn't match the ones in our Stripe account. And we checked all of our other Stripe accounts as well and none of them have the keys mentioned in Stripe console?
Can you please look into this on priority. We've been on hold to launch since last 10 days literally only because of this bug

**Reply by logeshswell** [reply]
I will take a look right away and will keep you posted at the earliest.

**Reply by virajsoni06** [reply]
@logeshswell what's the update on this?

**Reply by logeshswell** [reply]
We tested the Stripe integration in multiple production stores and it's working fine as expected in both test and live modes.
So, it would be really helpful to debug your issue if you could share a screen recording of the issue to support@swell.is

Please drop a text in this thread once you sent a mail. I will take a look.

**Reply by virajsoni06** [reply]
@logeshswell sent email

**Reply by logeshswell** [reply]
I gone through the recording and it is clear that you are connecting with the right Stripe account as it live_account_id matches with the value in Swell system while live_publishable_key and live_secret_key doesn't match.

Try exploring the stripe oauth apps/tokens where the key should exist.

**Reply by logeshswell** [reply]
**Immediate/possible workaround:**
* Replace the publishable key and secret key from the Stripe dashboard in thee Swell system using PUT request.
* Try deleting the expired keys under Stripe oauth tokens (**Highly recommended**)

**Reply by virajsoni06** [reply]
@logeshswell there aren't any apps installed on Stripe, doubt oauth exists.
Want to understand why is Swell not picking up the right keys?

<img width="577" alt="Screenshot 2024-01-30 at 8 38 12 AM" src="https://github.com/swellstores/community/assets/13236286/ae765a8b-7215-4b2d-9e46-249c58e93a38">

**Reply by virajsoni06** [reply]
Also, another issue, after changing the Stripe keys manually, the order will go through but it creates 2 different payment requests in Stripe of which one is always "incomplete", while its not a customer related issue its quite concerning issue for our accounting and analytics. Why is it creating multiple payment requests in the same checkout flow?

**Reply by logeshswell** [reply]
> there aren't any apps installed on Stripe, doubt oauth exists.

Can you check the list of publishable keys present in stripe oauth and delete the one which is causing issue?

> Want to understand why is Swell not picking up the right keys?

I also don't have any clue as Swell don't have the privilege to choose a working version of keys. Instead, Stripe returns the pair of publishable and secret keys as part of Oauth and we save it under the stripe settings in your swell store. Those keys were used to make Stripe API calls.

To give you more clarity, please refer to this doc: https://stripe.com/docs/connect/oauth-reference#post-token-response

So, we will check with the Stripe team regarding this issue and keep you posted on this.

**Reply by virajsoni06** [reply]
@logeshswell not clear how can we get the publishable keys, it needs auth code from the app but there is no Swell app connected when checking in Stripe.

**Reply by virajsoni06** [reply]
@logeshswell?

**Reply by logeshswell** [reply]
@virajsoni06 Can you give us some time to investigate and respond to you with further information?

**Reply by virajsoni06** [reply]
@logeshswell sure, can you give an eta on it. We need to understand how long this issue will take to get it resolved

**Reply by logeshswell** [reply]
> Also, another issue, after changing the Stripe keys manually, the order will go through.

I hope the integration issue is resolved for the time-being as per your comment [here](https://github.com/orgs/swellstores/discussions/478#discussioncomment-8295665)

Secondly, we already filed a bug ticket stating that double payment intents are created for googlePay method in Stripe. 
It is being worked on by our engineers and will be shipped quickly.

Is there anything else do you need assistance?

**Reply by virajsoni06** [reply]
@logeshswell I wouldn't call it resolved yet. The issue still exists where we do not know where the api keys are coming from and what is suggested is a makeshift solution. Either Swell should allow us to fully control the integration or it should be fixed asap. This is fairly serious issue and we won't call it as resolved at all. 

Firstly, can you please clearly and technically let me know where are the api keys coming from? And how can we access them? (PS: Sending a link to Stripe oauth does not help, need to clearly understand where are the keys from)

Secondly, the double payment intents does not exist for Google Pay, infact we do not have Google Pay enabled at all. It happens with credit/debit card payments, we never said its for Google Pay. Why is it happening for normal payment methods? Was the Stripe integration tested?

**Reply by logeshswell** [reply]
@virajsoni06 I also mean that the makeshift solution helps you for the time being. We will investigate and make a permanent fix for your Stripe integration issue.

> the double payment intents issue happens with credit/debit card payments.

Yes, I am aware of it and it happens with all payment methods. We are working on it.

**Comment by steven-k1** [comment, marked as answer]
@virajsoni06 A fix was released for the double payment intents.

**Reply by virajsoni06** [reply]
Hi @steven-k1 what about the original issue where the keys on Swell are different than that on Stripe?

**Reply by virajsoni06** [reply]
Don't understand why is this being closed without resolving the original issue

**Reply by logeshswell** [reply]
> what about the original issue where the keys on Swell are different than that on Stripe?

We will keep you posted in this same discussion after the fix is rolled out.

**Reply by logeshswell** [reply]
> Don't understand why is this being closed without resolving the original issue

If there's no further topic to discuss, we will close the discussion after a week or two. 
But, it doesn't mean that your reported issues were resolved, while closing the discussion.

You can always feel free to reopen the discussion for any other information.
